### PR TITLE
Feature declaring purpose of each file in file transfer token

### PR DESCRIPTION
**Pull request by rwrzesien**
Resolves https://github.com/golemfactory/concent/issues/355

### CODE REVIEW

File: concent_api/gatekeeper/tests/test_gatekeeper_views.py [hunk 1]
```diff
@@ -351,6 +351,42 @@ def test_upload_should_return_401_if_specific_file_info_data(self):
         self.assertIn('message', response.json().keys())
         self.assertEqual("application/json", response["Content-Type"])
 
+    @freeze_time("2018-12-30 11:00:00")
+    def test_upload_should_return_401_if_file_category_is_not_unique(self):
```
Comment (dybi) [inline concent_api/gatekeeper/tests/test_gatekeeper_views.py]: hm... maybe better repfrase it to sth like `...if_flie_category_is_the_same_for_both_files` or `..._not_unique_among_two_files`

even better ;)
`_not_unique_across_file_info_list`

File: concent_api/gatekeeper/tests/test_gatekeeper_views.py [hunk 2]
```diff
@@ -681,3 +717,39 @@ def test_download_should_return_401_concent_auth_header_is_not_loadable(self):
         self.assertIsInstance(response, JsonResponse)
         self.assertEqual(response.status_code, 401)
         self.assertIn('message', response.json().keys())
+
+    @freeze_time("2018-12-30 11:00:00")
+    def test_download_should_return_401_if_file_category_is_not_unique(self):
```
Comment (dybi) [inline concent_api/gatekeeper/tests/test_gatekeeper_views.py]: same here

File: concent_api/gatekeeper/views.py [hunk 3]
```diff
@@ -353,6 +353,39 @@ def parse_headers(request: WSGIRequest, path_to_file: str) -> Union[FileTransfer
...
+    ):
+        return gatekeeper_access_denied_response(
+            "'category' field has invalid value.",
+            ErrorCode.MESSAGE_FILES_CATEGORY_INVALID,
+            path_to_file,
+            loaded_golem_message.subtask_id,
+            concent_client_public_key
+        )
+
+    if not (
```
Comment (dybi) [inline concent_api/gatekeeper/views.py]: this validation is incorrect: 
for example, if you change the test `test_upload_should_return_401_if_file_category_is_not_unique` in a way that it contains files with two different categories, the test will still pass.

Please change it to sth like:
```python
categories = [file_info['category'] for file_info in loaded_golem_message.files]
    if len(set(categories)) != len(categories):
```

File: concent_api/gatekeeper/tests/test_gatekeeper_views.py (hunk 1 above)
Comment (rwrzesien) [reply concent_api/gatekeeper/tests/test_gatekeeper_views.py]: :+1: done

File: concent_api/gatekeeper/tests/test_gatekeeper_views.py (hunk 2 above)
Comment (rwrzesien) [reply concent_api/gatekeeper/tests/test_gatekeeper_views.py]: :+1: done

File: concent_api/gatekeeper/views.py (hunk 3 above)
Comment (rwrzesien) [reply concent_api/gatekeeper/views.py]: > if you change the test test_upload_should_return_401_if_file_category_is_not_unique in a way that it contains files with two different categories, the test will still pass.

That is correct behavior.

**Comment by dybi** [comment]
After rebase some tests fail:
- `test_that_concent_accepts_valid_request_and_sends_verification_order_to_work_queue` (F)
- `test_download_should_return_401_if_file_categories_are_not_unique_across_file_info_list` (E)
- `test_upload_should_return_401_if_file_categories_are_not_unique_across_file_info_list` (E)

File: concent_api/gatekeeper/views.py (hunk 3 above)
Comment (dybi) [reply concent_api/gatekeeper/views.py]: what I ment is:
```python
    def test_upload_should_return_401_if_file_categories_are_not_unique_across_file_info_list(self):
        file_1 = FileTransferToken.FileInfo(
            path='blender/benchmark/test_task/scene-Helicopter-27-cycles.blend',
            checksum='sha1:95a0f391c7ad86686ab1366bcd519ba5ab3cce89',
            size=1024,
            category=FileTransferToken.FileInfo.Category.results
        )
        file_2 = FileTransferToken.FileInfo(
            path='blender/benchmark/test_task/scene-Helicopter-28-cycles.blend',
            checksum='sha1:95a0f391c7ad86686ab1366bcd519ba5ab3cce89',
            size=1024,
            category=FileTransferToken.FileInfo.Category.results
        )
```
produces: `MESSAGE_FILES_CATEGORY_NOT_UNIQUE`,
but if you change it to:
```python
    def test_upload_should_return_401_if_file_categories_are_not_unique_across_file_info_list(self):
        file_1 = FileTransferToken.FileInfo(
            path='blender/benchmark/test_task/scene-Helicopter-27-cycles.blend',
            checksum='sha1:95a0f391c7ad86686ab1366bcd519ba5ab3cce89',
            size=1024,
            category=FileTransferToken.FileInfo.Category.results
        )
        file_2 = FileTransferToken.FileInfo(
            path='blender/benchmark/test_task/scene-Helicopter-28-cycles.blend',
            checksum='sha1:95a0f391c7ad86686ab1366bcd519ba5ab3cce89',
            size=1024,
            category=FileTransferToken.FileInfo.Category.resources
        )
```
(so now `FileTransferToken` has files belonging to two **different** categories), you still get:
`MESSAGE_FILES_CATEGORY_NOT_UNIQUE`. Please check it and fix validation

Comment (rwrzesien) [reply concent_api/gatekeeper/views.py]: :+1: done

**Pull request merged**